src/main/rx/jetiexbus.c

```diff
@@ -0,0 +1,595 @@
+} 
+
+// Jeti Ex Telemetry CRC calculations for a frame
+uint8_t calcCRC8(uint8_t *pt, uint8_t msgLen)
+{
+    uint8_t crc8_data = 0;
+    uint8_t mlen = 0;
+    for (mlen = 0; mlen < msgLen; mlen++) {
+        crc8_data = updateCRC8(crc8_data, pt[mlen]);
+    }
+    return(crc8_data);
+}
+
+#endif //TELEMETRY
+
+
+void jetiExBusDecodeFrame(uint8_t *pFrame)
+{
+    uint16_t value;
+    uint8_t frameAddr;
+
+    // Decode header
+    switch (((uint16_t)jetiExBusFrame[EXBUS_HEADER_SYNC] << 8) + ((uint16_t)jetiExBusFrame[EXBUS_HEADER_REQ])){
+
+    case EXBUS_CHANNELDATA_TELEMETRY_REQUEST:                   // not yet specified
+    case EXBUS_CHANNELDATA:
+        for (uint8_t i = 0; i < JETIEXBUS_CHANNEL_COUNT; i++) {
+            frameAddr = JETIEXBUS_HEADER_LEN + i * 2;
+            value = ((uint16_t)pFrame[frameAddr + 1]) << 8;
+            value += (uint16_t)pFrame[frameAddr];
+            // Convert to internal format
+            jetiExBusChannelData[i] = value >> 3;
+        }
+        break;
+    }
+}
+
+void jetiExBusFrameReset(void)
+{
+    jetiExBusFramePosition = 0;
+    jetiExBusFrameLength = JETIEXBUS_MAX_FRAME_SIZE;
+}
+
+
+/*
+  supported:
+  0x3E 0x01 LEN Packet_ID 0x31 SUB_LEN Data_array CRC16      // Channel Data with telemetry request (2nd byte 0x01)
+  0x3E 0x03 LEN Packet_ID 0x31 SUB_LEN Data_array CRC16      // Channel Data forbids answering (2nd byte 0x03)
+  0x3D 0x01 0x08 Packet_ID 0x3A 0x00 CRC16                   // Telemetry Request EX telemetry (5th byte 0x3A)
+  
+  other messages - not supported:
+  0x3D 0x01 0x09 Packet_ID 0x3B 0x01 0xF0 CRC16              // Jetibox request (5th byte 0x3B)
+  ...
+  
+*/
+// Receive ISR callback
+static void jetiExBusDataReceive(uint16_t c)
+{
+    uint32_t now;
+    static uint32_t jetiExBusTimeLast = 0;
+    static uint32_t jetiExBusTimeInterval;
+    static uint16_t crc = 0;
+    
+    // Check if we shall reset frame position due to time
+    now = millis();
+
+    jetiExBusTimeInterval = now - jetiExBusTimeLast;
+    jetiExBusTimeLast = now;
+
+    if (jetiExBusTimeInterval > JETIEXBUS_MIN_FRAME_GAP) {
+        jetiExBusFrameReset();
+        jetiExBusFrameState = EXBUS_STATE_ZERO;
+        jetiExBusRequestState = EXBUS_STATE_ZERO;
+    }
+    
+    // Check if we shall start a frame?
+    if (jetiExBusFramePosition == 0) {
+        switch(c){
+        case JETIEXBUS_START_CHANNEL_FRAME:
+            jetiExBusFrameState = EXBUS_STATE_IN_PROGRESS;
+            jetiExBusFrame = &jetiExBusSerialFrame[0];
+            break;
+            
+        case JETIEXBUS_START_REQUEST_FRAME:
+            jetiExBusRequestState = EXBUS_STATE_IN_PROGRESS;
+            jetiExBusFrame = &jetiExBusSerialFrame[JETIEXBUS_MAX_FRAME_SIZE];
+            break;
+
+        default:
+            return;
+        }
+        crc = 0;
+    }
+
+    // Store in frame copy
+    jetiExBusFrame[jetiExBusFramePosition] = (uint8_t)c;
+    crc = updateCRC16(crc, (uint8_t)c);
```

Comment:
CRC may be calculated when processing packet, saving some time in IRQ handler (and also some time in general, crc loop can be optimized)